airgun/views/rhai.py

```diff
@@ -0,0 +1,138 @@
+from widgetastic_patternfly import Button, Dropdown
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.widget import (
+    Checkbox,
+    GenericLocatorWidget,
+    ParametrizedView,
+    Table,
+    Text,
+    TextInput,
+    View
+)
+
+from airgun.views.common import BaseLoggedInView
+
+
+class Overview(BaseLoggedInView):
+    title = Text(".//h1[@class='page-title']//span")
+
+    @View.nested
+    class action_summary(View):
+        title = Text(".//h2[@class='title']")
+
+    @View.nested
+    class newest_systems(View):
+        title = Text(".//h2[@class='title']")
+        table = Table(".//section[contains(normalize-space(.),"
+                      "'Newest Systems')]//table")
+
+    @View.nested
+    class optimize_your_expeirience(View):
+        title = Text(".//h2[@class='title']")
+        registered_systems = Text(".//div[@class='chart-metrics']/span")
+        stability_issues = Text(
+            ".//span[normalize-space(.)='View stability issues']"
+            "/../../div[contains(@class, 'count')]")
+        security_issues = Text(
+            ".//span[normalize-space(.)='View security issues']"
+            "/../../div[contains(@class, 'count')]")
+
+
+class Actions(BaseLoggedInView):
+    title = Text(".//h1[@class='page-title']//span")
+    issues = Text(".//div[@class='num']")
+    export_csv = Button("Export CSV")
+
+
+class InventoryAllHosts(BaseLoggedInView):
+    title = Text(".//h1[@class='page-title']//span")
+    search = TextInput(locator=".//input[@placeholder='Find a system']")
+    actions = Dropdown("Actions")
+    systems_count = Text(".//h3[@class='system-count']")
+    table = Table(".//table",
+                  column_widgets={0: Checkbox(locator=".//input")})
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Inventory" and
+            self.menu.currently_selected == ["Insights", "Inventory"]
+        )
+
+
+class InventoryHostDetails(View):
+    hostname = Text(".//div[@class='modal-title']/h2//span")
```

Comment:
any other way to address that element? We try to avoid absolute paths

Reply:
It's within a modal window.
```html
<div class="system-overlay">
	<div class="modal-actions">
		<div ng-click="close()" class="fa fa-close pull-right" role="button" tabindex="0"></div>
	</div>
	<div class="modal-title">
		<div class="icon">
			<span class="fa fa-globe"></span>
		</div><!----><!----><!-- dont show hostname if it is not avalible--><!---->
		<h2 ng-if="system.hostname &amp;&amp; !system.display_name">
			<b translate="translate">
				<span>hostname:</span>
			</b>
			<div class="ellipsis-overflow">
				<span>37hall10eqxpemxe.lab.eng.rdu2.redhat.com</span>
			</div>
		</h2><!----><!-- always show the uuid.... it must exist-->
		<div class="ellipsis-overflow">
			<div class="small-name-info">UUID: 1829eab7-e57e-43a9-92b6-bca3c6c00809</div>
		</div>
	</div>
</div>
```